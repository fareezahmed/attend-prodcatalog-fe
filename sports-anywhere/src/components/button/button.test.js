import React from 'react';
import { mount } from 'enzyme'

import Button from './index';


const setup = (partialProps) => {
    const props = {
        childern: "Facbook",
        ...partialProps,
      }
    const wrapper = mount(<Button { ...props } />);
    return {
        wrapper,
        props,
    }
}

describe('Button component', () => {
    it('Default button component renders correctly', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    })

    it('Primary button component renders correctly', () => {
        const variation = "primary";
        const { wrapper } = setup({variation});
        expect(wrapper).toMatchSnapshot();
    })

    it('Primary button component renders correctly', () => {
        const variation = "secondary";
        const { wrapper } = setup({variation});
        expect(wrapper).toMatchSnapshot();
    })

})