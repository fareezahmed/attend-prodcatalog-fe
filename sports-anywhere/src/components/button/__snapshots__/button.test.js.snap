// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Default button component renders correctly 1`] = `
<Button>
  <button
    className="btn btn--document"
  >
    Facbook
  </button>
</Button>
`;

exports[`Button component Primary button component renders correctly 1`] = `
<Button
  variation="primary"
>
  <button
    className="btn btn--primary"
  >
    Facbook
  </button>
</Button>
`;

exports[`Button component Primary button component renders correctly 2`] = `
<Button
  variation="secondary"
>
  <button
    className="btn btn--secondary"
  >
    Facbook
  </button>
</Button>
`;
